src/index.js

```diff
@@ -313,145 +289,174 @@ function addSegment(currentState, segment) {
 
 // The main interface
 
-export var RouteRecognizer = function() {
-  this.rootState = new State();
-  this.names = {};
-};
-
-
-RouteRecognizer.prototype = {
-  add: function(routes, options) {
-    var currentState = this.rootState, regex = "^",
-        types = { statics: 0, dynamics: 0, stars: 0 },
-        handlers = [], allSegments = [], name;
-
-    var isEmpty = true;
-
-    for (var i=0, l=routes.length; i<l; i++) {
-      var route = routes[i], names = [];
-
-      var segments = parse(route.path, names, types);
+export class RouteRecognizer {
+  constructor() {
+    this.map = map;
+    this.rootState = new State();
+    this.names = {};
+  }
 
-      allSegments = allSegments.concat(segments);
+  add(route) {
```

Comment:
The previous implementation (sort of) supported multiple routes here, but didn't correctly register them all. All code paths I could find only ever pass an array with a single route.